script.js

```diff
@@ -1,4 +1,255 @@
+  while (counter < numberOfDice) {
+    diceListRollResults.push(`Die ${counter + 1}: ${diceArray[counter]}`);
+    counter = counter + 1;
+  }
+
+  //Sorts the dice array in descending order to create the player's optimal dice combo.
+  diceComboArray = diceArray.sort(function (a, b) {
+    return b - a;
+  });
+
+  //converts the dice combo number into an integer, then places it into the respective player's variable for retriving later
+  if (playerTurn == 1) {
+    p1DiceCombo = Number(diceComboArray.join(""));
+  } else if (playerTurn == 2) {
+    p2DiceCombo = Number(diceComboArray.join(""));
+  } else if (playerTurn == 3) {
+    p3DiceCombo = Number(diceComboArray.join(""));
+  } else if (playerTurn == 4) {
+    p4DiceCombo = Number(diceComboArray.join(""));
+  }
+
+  myOutputValue = `PLAYER ${playerTurn}'S ROLLS <br><br> ${diceListRollResults.join(
+    "<br>"
+  )} <br><br> Your number is ${diceComboArray.join("")} <br> <br> Player ${
+    playerTurn + 1
+  }, click the button to roll.`;
+
+  if (playerTurn == numberOfPlayers) {
+    //at the last player's roll, change to display results mode.
+    gameState = "displayResults";
+
+    myOutputValue = `PLAYER ${playerTurn}'S ROLLS <br><br> ${diceListRollResults.join(
+      "<br>"
+    )} <br><br> Your number is ${diceComboArray.join(
+      ""
+    )} <br> <br>  Click the button to see the results for the round.`;
+  }
+
+  playerTurn = playerTurn + 1;
   return myOutputValue;
 };
+
+//EXECUTE SELECT AUTO -  used in the 'auto-generate combined number' section
+/*
+var executeSelectAuto = function (input) {
+  if (die1 > die2) {
+    diceCombo = Number(`${die1}` + `${die2}`);
+  }
+  if (die2 > die1) {
+    diceCombo = Number(`${die2}` + `${die1}`);
+  }
+  if (die1 == die2) {
+    diceCombo = Number(`${die1}` + `${die2}`);
+  }
+
+  //assign the results to respective players
+  if (playerTurn == 2) {
+    diceComboP2 = diceCombo;
+    gameState = "displayResults";
+    playerTurn = 1;
+    player2Score = player2Score + diceCombo;
+  } else if (playerTurn == 1) {
+    diceComboP1 = diceCombo;
+    gameState = "roll";
+    playerTurn = 2;
+    player1Score = player1Score + diceCombo;
+  }
+
+  return selectMessage();
+}; */
+
+//EXECUTE DISPLAY RESULTS
+var executeDisplayResults = function (input) {
+  //resets the game state to "roll" and starts again from player 1.
+  gameState = "roll";
+  playerTurn = 1;
+  roundNumber = roundNumber + 1;
+
+  return displayNewResultsMessage();
+};
+
+var main = function (input) {
+  if (gameState == "preGame") {
+    return executePreGame(input);
+  } else if (gameState == "chooseNumberOfPlayers") {
+    return chooseNumberOfPlayers(input);
+  } else if (gameState == "chooseDiceNumber") {
+    return chooseDiceNumber(input);
+  } else if (gameState == "roll") {
+    return executeNewRoll(input);
+  } else if (gameState == "displayResults") {
+    return executeDisplayResults(input);
+  }
+};
```

Comment:
i like this part of the main function that you did , making it look clean and easy to understand.